erpnext/patches/v12_0/update_end_date_and_status_in_email.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import unicode_literals
+import frappe
+from frappe.utils import add_days, getdate, today
+
+def execute():
+    email_campaign = frappe.get_all('Email Campaign')
+    if not email_campaign:
+        return
+    for campaign in email_campaign:
+        doc = frappe.get_doc("Email Campaign",campaign["name"])
+        send_after_days = []
+
+        camp = frappe.get_doc("Campaign", doc.campaign_name)
+        for entry in camp.get("campaign_schedules"):
+            send_after_days.append(entry.send_after_days)
+        if send_after_days:
+            end_date = add_days(getdate(doc.start_date), max(send_after_days))
+            doc.db_set("end_date", end_date)
+        today_date = getdate(today())
+        if doc.start_date > today_date:
+            doc.db_set("status", "Scheduled")
+        elif end_date >= today_date:
+            doc.db_set("status", "In Progress")
+        elif end_date < today_date:
+            doc.db_set("status", "Completed")
+        frappe.db.commit()
```

Comment:
Commit is not required. It can lead to committing partially if there are exceptions after it is executed.